### PR TITLE
Convert api-browser and api-node docs to TS

### PR DESCRIPTION
**Pull request by gabrieluizramos**
## Description
In order to continue the conversion, this PR converts `api-browser-docs` and `api-node-docs` to TS.

## Related Issues

Related to #21995

### CODE REVIEW

**Comment by gabrieluizramos** [comment]
@blainekasten Hi Blaine! Can you help me with this one? All the tests are passing locally but there's one check with error that I can't access...

File: packages/gatsby/src/utils/api-node-docs.ts [hunk 1]
```diff
@@ -0,0 +1,418 @@
...
+ *     },
+ *   });
+ * }
+ */
+export const onCreateWebpackConfig = true
+
+/**
+ * The first API called during Gatsby execution, runs as soon as plugins are loaded, before cache initialization and bootstrap preparation.
+ */
+export const sonPreInit = true
```
Comment (ascorbic) [inline packages/gatsby/src/utils/api-node-docs.ts:375]: Typo

Comment (gabrieluizramos) [reply packages/gatsby/src/utils/api-node-docs.ts]: Thanks! I'm pushing the fix right now!

**Comment by dja** [comment]
It looks like the production page for the Gatsby Browser APIs is now blank. Is it related to this?

**Comment by gabrieluizramos** [comment]
@dja not sure, but I've already opened the revert so we can test it: https://github.com/gatsbyjs/gatsby/pull/23853

**Comment by ascorbic** [comment]
I have a fix for the API docs #23863

**Pull request merged**